app/models/procedure.rb

```diff
@@ -130,4 +130,23 @@ def total_dossier
     self.dossiers.where.not(state: :draft).size
   end
 
+  def export_full_generation
+    data = []
+    headers = []
+
+    exportable_dossiers = dossiers.downloadable
+
+    if exportable_dossiers && exportable_dossiers.any?
+      headers = exportable_dossiers.first.export_headers
+      dossiers.each do |dossier|
```

Comment:
On peut faire mieux :

- on enlève
```ruby
data = []
headers = []
```

- on transforme le code d'itération sur dossiers :
```ruby
data = dossiers.map do |dossier|
  dossier.convert_specific_array_values_to_string(dossier.data_with_champs)
end
```

Note : je veux bien que tu fasses une autre PR, à la suite ce celle-ci, qui refactor `convert_specific_array_values_to_string` :
- c'est appelé qu'ici, donc pas besoin de passer dossier.data_with_champs, on peut appeler ça directement dans la méthode
- on peut du coup renommer la méthode en quelque chose de plus clair